pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -156,9 +149,21 @@ autoPatchelf() {
 
     # Add all shared objects of the current output path to the start of
     # cachedDependencies so that it's choosen first in findDependency.
-    cachedDependencies+=(
-        $(find "$prefix" \! -type d \( -name '*.so' -o -name '*.so.*' \))
-    )
+    # This error is ignored, because $prefix is not defined in this script.
+    # shellcheck disable=SC2154
+    while IFS='' read -r -d $'\0' file; do
+        cachedDependencies+=("$file")
+    done < <(find "$prefix" \! -type d \( -name '*.so' -o -name '*.so.*' \) -print0)
```

Comment:
Again here, I think this makes the code less readable.

Reply:
The reason for the while loop is `-print0` which was suggested above. I need to use this way (`done < <(`) instead of `find | while`, because the latter one spawns a subshell and so I'm not able to modify variables (see [SC2030](https://github.com/koalaman/shellcheck/wiki/SC2030))

Reply:
I thought that was fixed recently...?

Reply:
Oh was it??